app/api/sections.js

```diff
@@ -0,0 +1,62 @@
+const { Router } = require('express')
+const router = Router()
+
+const db = require('../models/index')
+const { logger } = require('../../lib/logger')
+const sectionService = require('../services/section_service')
+const { requireAuthentication, generateOTP } = require('../../lib/auth')
+const { ValidationError, UniqueConstraintError } = require('sequelize')
+
+router.post('/:course_id/sections', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    // make sure the user creating this section is the teacher for the course
+    const enrollment = await db.Enrollment.findOne({
+        where: { 
+            userId: user.id,
+            courseId: parseInt(req.params['course_id']),
+            role: 'teacher'
+        }
+    })
+
+    if (req.body.number && enrollment) {   // req.body.number not required as it will be auto incremented if not present
```

Comment:
I'm a little confused because it errors if the number is not present. I would just do 1 if statement checking for enrollment, then make sure you filter the body on the create request - number being one of the fields that your filter includes but obviously not required like you said

Reply:
I just forgot to uncomment that comment, if you remember when we talked about this it turns out number is required